pkg/operation/seed/seed.go

```diff
@@ -326,23 +328,52 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
 		return err
 	}
 
+	// HVPA feature gate
+	hvpaEnabled := gardenletfeatures.FeatureGate.Enabled(features.HVPA)
+	if !hvpaEnabled {
+		if err := common.DeleteHvpa(ctx, k8sSeedClient, v1beta1constants.GardenNamespace); client.IgnoreNotFound(err) != nil {
+			return err
+		}
+	}
+
 	// Logging feature gate
 	var (
 		loggingEnabled                    = gardenletfeatures.FeatureGate.Enabled(features.Logging)
 		existingSecretsMap                = map[string]*corev1.Secret{}
 		filters                           = strings.Builder{}
 		parsers                           = strings.Builder{}
 		fluentBitConfigurationsOverwrites = map[string]interface{}{}
+		lokiValues                        = map[string]interface{}{}
 	)
 
+	lokiValues["enabled"] = loggingEnabled
+
 	if loggingEnabled {
-		// TODO: remove in a future release
-		// Clean up the stale loki-vpa.
+		lokiValues["authEnabled"] = false
+		lokiValues["hvpa"] = map[string]interface{}{
+			"enabled": hvpaEnabled,
+		}
+
 		lokiVpa := &autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "loki-vpa", Namespace: v1beta1constants.GardenNamespace}}
 		if err := k8sSeedClient.Client().Delete(ctx, lokiVpa); client.IgnoreNotFound(err) != nil {
 			return err
 
 		}
+
+		if hvpaEnabled {
+			foundLokiStatefulset := true
+			lokiStatefulSet := &appsv1.StatefulSet{}
+			if err := k8sSeedClient.Client().Get(ctx, kutil.Key(v1beta1constants.GardenNamespace, "loki"), lokiStatefulSet); client.IgnoreNotFound(err) != nil {
+				return err
+			} else if apierrors.IsNotFound(err) {
+				foundLokiStatefulset = false
+			}
+
+			if foundLokiStatefulset && hvpaEnabled {
+				lokiValues["resources"] = lokiStatefulSet.Spec.Template.Spec.Containers[0].Resources.DeepCopy()
+			}
```

Comment:
Code duplication. What about extraction this into a function and calling it on the two places that use it?